sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -249,6 +249,15 @@ class JDBCOptions(
       .map(_.toBoolean)
       .getOrElse(SQLConf.get.timestampType == TimestampNTZType)
 
+  val hint = {
+    parameters.get(JDBC_HINT_STRING).map(value => {
+      require(value.startsWith("/*+ ") && value.endsWith(" */"),
```

Comment:
Oh, what I expected was a regex instead of this, @wayneguow .

Reply:
Updated it.